### PR TITLE
added book logic

### PR DESCRIPTION
**Pull request by Oleksii-Melashchenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -68,26 +83,13 @@
...
-                            <artifactId>lombok</artifactId>
-                        </exclude>
-                    </excludes>
-                </configuration>
-            </plugin>
         </plugins>
     </build>
 
-</project>
+</project>
```
Comment (liliia-ponomarenko) [inline pom.xml]: add one empty line at the end of the file

File: src/main/java/com/clozex/shop/ShopApplication.java [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class ShopApplication {
 
+    private final BookService bookService;
+
+    @Autowired
+    public ShopApplication(BookService bookService) {
+        this.bookService = bookService;
```
Comment (liliia-ponomarenko) [inline src/main/java/com/clozex/shop/ShopApplication.java]: Let’s  use RequiredArgsConstructor

File: src/main/java/com/clozex/shop/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
```
Comment (liliia-ponomarenko) [inline src/main/java/com/clozex/shop/service/impl/BookServiceImpl.java]: the same

File: src/main/java/com/clozex/shop/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    @Override
+    public Book save(Book book) {
+        return bookRepository.save(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return bookRepository.findAll();
+    }
+
```
Comment (liliia-ponomarenko) [inline src/main/java/com/clozex/shop/service/impl/BookServiceImpl.java]:
```suggestion
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  fix a few mistakes ;)

File: src/main/java/com/clozex/shop/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/clozex/shop/model/Book.java]: don't use `@Data` for entities
```suggestion
@Getter
@Setter
```

File: src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+import com.clozex.shop.model.Book;
+import com.clozex.shop.repository.BookRepository;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java:13]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private SessionFactory sessionFactory;
+
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
    private final SessionFactory sessionFactory;
```

File: src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can`t save book" + book, e);
```
Comment (Elena-Bruyako) [inline src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can`t save book" + book, e);
```

File: src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM Book ", Book.class)
+                    .getResultList();
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/com/clozex/shop/repository/impl/BookRepositoryImpl.java:47]: add catch block

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**